File: 238 Product of Array Except Self/238 Product of Array Except Self.ts
function productExceptSelf(nums: number[]): number[] {
  const result: number[] = Array(nums.length).fill(1);
  let pre: number = 1;
  let post: number = 1;

  for (let i = 0; i < nums.length; i++) {
    result[i] = pre;
    pre *= nums[i];
  }

  for (let i = nums.length - 1; i >= 0; i--) {
    result[i] *= post;
    post *= nums[i];
  }

  return result;
}

File: 238 Product of Array Except Self/238 Product of Array Except Self.swift
class Solution {
  func productExceptSelf(_ nums: [Int]) -> [Int] {
    var result: [Int] = Array(repeating: 1, count: nums.count)
    var pre: Int = 1
    var post: Int = 1

    for i in 0..<nums.count {
      result[i] = pre
      pre *= nums[i]
    }

    for i in stride(from: nums.count - 1, through: 0, by: -1) {
      result[i] *= post
      post *= nums[i]
    }

    return result
  }
}
